enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/APIKeyAuthenticator.java

```diff
@@ -0,0 +1,513 @@
+                                .toLowerCase());
+                        requestContext.addResponseHeaders(APIConstants.AUTHORIZATION_HEADER_DEFAULT,
+                                APIConstants.AUTHORIZATION_HEADER_BASIC + ' ' +
+                                        Base64.getEncoder().encodeToString((securityInfo.getUsername() +
+                                                ':' + securityInfo.getPassword()).getBytes()));
+                    }
+                }
+
+                validateAPIKeyRestrictions(payload, requestContext, apiContext, apiVersion);
+
+                validateAPISubscription(apiContext, apiVersion, payload, splitToken, false);
+
+                log.debug("API Key authentication successful.");
+
+                // Begins analytics data processing
+
+                //Get SignedJWTInfo
+                SignedJWTInfo signedJWTInfo = getSignedJwt(apiKey);
+
+                //Get JWTValidationInfo
+                JWTValidationInfo validationInfo = new JWTValidationInfo();
+                validationInfo.setUser(payload.getSubject());
+
+                String endUserToken = null;
+
+                // Get jwtConfigurationDto
+                JWTConfigurationDto jwtConfigurationDto = ConfigHolder.getInstance().
+                        getConfig().getJwtConfigurationDto();
+                if (jwtConfigurationDto.isEnabled()) {
+                    // Set ttl
+                    jwtConfigurationDto.setTtl(JWTUtil.getTTL());
+
+                    JWTInfoDto jwtInfoDto = FilterUtils
+                            .generateJWTInfoDto(null, validationInfo, apiKeyValidationInfoDTO, requestContext);
+                    endUserToken = generateAndRetrieveJWTToken(tokenIdentifier, jwtInfoDto);
+                    // Set generated jwt token as a response header
+                    requestContext.addResponseHeaders(jwtConfigurationDto.getJwtHeader(), endUserToken);
+                }
+                JWTClaimsSet claims = signedJWTInfo.getJwtClaimsSet();
+
+                AuthenticationContext authenticationContext = FilterUtils
+                        .generateAuthenticationContext(requestContext, tokenIdentifier, validationInfo,
+                                apiKeyValidationInfoDTO, endUserToken, false);
+
+                if (claims.getClaim("keytype") != null) {
+                    authenticationContext.setKeyType(claims.getClaim("keytype").toString());
+                }
+                log.debug("Analytics data processing for API Key (jiti) " + tokenIdentifier +
+                        " was successful");
+                return authenticationContext;
+
+            }
+        } catch (ParseException e) {
+            log.debug("API Key authentication failed. ", e);
+        }
+
+
+        throw new APISecurityException(APIConstants.StatusCodes.UNAUTHENTICATED.getCode(),
+                APISecurityConstants.API_AUTH_GENERAL_ERROR, APISecurityConstants.API_AUTH_GENERAL_ERROR_MESSAGE);
```

Comment:
Do we need to throw from here as well. I don't think the execution path comes to this line.